### PR TITLE
Added version control delete branch and hard reset checkpoints documentation

### PR DESCRIPTION
**Pull request by yaustar**
Fixes #https://github.com/playcanvas/developer.playcanvas.com/issues/292 and https://github.com/playcanvas/developer.playcanvas.com/issues/289

### CODE REVIEW

**Comment by willeastcott** [comment]
How come this PR includes to safe area docs?

**Comment by yaustar** [comment]
> How come this PR includes to safe area docs?

I screwed up my branching :( 

Now resolved :)

File: content/en/user-manual/version-control/branches.md [hunk 1]
```diff
@@ -36,16 +36,37 @@ The editor will reload with your current branch switched to the chosen branch.
 
 ![Close branch][4]
 
-It is not possible to delete branches as they are required for future merges. However, if you have completed work on a branch you can close the branch which will remove it from the interface.
+However, if you have completed work on a branch you can close the branch which will remove it from the interface.
```
Comment (yaustar) [inline content/en/user-manual/version-control/branches.md:39]:
```suggestion
If you have completed work on a branch you can close the branch which will remove it from the the open branches list.
```

File: content/en/user-manual/version-control/checkpoints.md [hunk 2]
```diff
@@ -26,13 +26,36 @@ The New Checkpoint button opens the input form to create a checkpoint. You can a
...
 ![Restore checkpoint applied][5]
 
+## Hard reset to a checkpoint
+
+Hard reset allows you to delete all checkpoints after a selected checkpoint. This is useful if you need to 'undo' a merge (perhaps a branch was accidentally merged or the merge needs to be done differently).
+
+Hard reset can only delete the checkpoints if the following conditions are met:
+
+- No branches have been created from the checkpoints being deleted
+- The checkpoints being deleted has not been created by a merge of branches
```
Comment (yaustar) [inline content/en/user-manual/version-control/checkpoints.md:40]:
```suggestion
- The checkpoints being deleted have not been created by a merge of branches
```

File: content/en/user-manual/version-control/checkpoints.md [hunk 3]
```diff
@@ -26,13 +26,36 @@ The New Checkpoint button opens the input form to create a checkpoint. You can a
...
+
+**Note, deleted checkpoints cannot be recovered.**
+
+To hard reset, open the panel, find the checkpoint that you'd like to reset to to and choose "Hard reset" from the checkpoint's drop down menu.
+
+![][hard-reset]
+
+Type 'hard rest' in the text box to confirm that you wish to delete all checkpoints after the selected checkpoint.
```
Comment (yaustar) [inline content/en/user-manual/version-control/checkpoints.md:48]:
```suggestion
Type 'hard reset' in the text box to confirm that you wish to delete all checkpoints after the selected checkpoint.
```

**Comment by yaustar** [comment]
@willeastcott Could I have another pass please :)

File: content/en/user-manual/version-control/branches.md [hunk 4]
```diff
@@ -36,16 +36,37 @@ The editor will reload with your current branch switched to the chosen branch.
 
 ![Close branch][4]
 
-It is not possible to delete branches as they are required for future merges. However, if you have completed work on a branch you can close the branch which will remove it from the interface.
+If you have completed work on a branch you can close the branch which will remove it from the the open branches list.
```
Comment (willeastcott) [inline content/en/user-manual/version-control/branches.md:39]:
```suggestion
If you have completed work on a branch, you can close it which will remove it from the open branches list.
```

File: content/en/user-manual/version-control/branches.md [hunk 5]
```diff
@@ -36,16 +36,37 @@ The editor will reload with your current branch switched to the chosen branch.
...
 
-To close a branch open the version control panel, select the branch you wish to close and choose the "Close branch" option in the branch's drop-down menu. Note, you cannot close your current branch or the master branch. Switch to a different branch first if you wish to close your current branch.
+To close a branch open the version control panel, select the branch you wish to close and choose the "Close this branch" option in the branch's drop-down menu. Note, you cannot close your current branch or the master branch. Switch to a different branch first if you wish to close your current branch.
```
Comment (willeastcott) [inline content/en/user-manual/version-control/branches.md:41]:
```suggestion
To close a branch, open the version control panel, select the branch you wish to close and choose the "Close this branch" option in the branch's drop-down menu. Note that you cannot close your current branch or the master branch. Switch to a different branch first if you wish to close your current branch.
```

File: content/en/user-manual/version-control/branches.md [hunk 6]
```diff
@@ -36,16 +36,37 @@ The editor will reload with your current branch switched to the chosen branch.
...
-You will be asked to confirm the closing of the branch and you have an option to discard any changes that were made to your branch since you last took a checkpoint. By default, PlayCanvas will save your changes in an additional checkpoint before closing the branch. If you wish to discard these changes you can select the option here. **Note, enabling this checkbox will lose any work you have made in the branch since you last made a checkpoint**.
+You will be asked to confirm the closing of the branch and you have an option to discard any changes that were made to your branch since you last took a checkpoint. By default, PlayCanvas will save your changes in an additional checkpoint before closing the branch. If you wish to discard these changes you can select the option here.
+
+**Note, enabling this checkbox will lose any work you have made in the branch since you last made a checkpoint**.
+
+## Deleting a branch
+
+Deletion of branches are only supported if the following conditions are met:
+
+- The branch as not been merged into another branch
```
Comment (willeastcott) [inline content/en/user-manual/version-control/branches.md:53]:
```suggestion
- The branch has not been merged into another branch
```

File: content/en/user-manual/version-control/checkpoints.md [hunk 7]
```diff
@@ -26,13 +26,36 @@ The New Checkpoint button opens the input form to create a checkpoint. You can a
...
+
+Hard reset allows you to delete all checkpoints after a selected checkpoint. This is useful if you need to 'undo' a merge (perhaps a branch was accidentally merged or the merge needs to be done differently).
+
+Hard reset can only delete the checkpoints if the following conditions are met:
+
+- No branches have been created from the checkpoints being deleted
+- The checkpoints being deleted have not been created by a merge of branches
+
+**Note, deleted checkpoints cannot be recovered.**
+
```
Comment (willeastcott) [inline content/en/user-manual/version-control/checkpoints.md:43]:
```suggestion
To hard reset, open the panel, find the checkpoint that you'd like to reset to and choose "Hard reset" from the checkpoint's drop down menu.
```

File: content/en/user-manual/version-control/checkpoints.md [hunk 8]
```diff
@@ -26,13 +26,36 @@ The New Checkpoint button opens the input form to create a checkpoint. You can a
...
+To hard reset, open the panel, find the checkpoint that you'd like to reset to to and choose "Hard reset" from the checkpoint's drop down menu.
```
Comment (willeastcott) [inline content/en/user-manual/version-control/checkpoints.md:44]:
```suggestion
To hard reset, open the panel, find the checkpoint that you'd like to reset to and choose "Hard reset" from the checkpoint's drop down menu.
```

**Review by willeastcott — APPROVED** [review]
Approving with some suggestions. 😄

**Pull request merged**